### DISCUSSION TITLE
好久没来项目了，没想到，你居然添加了 DNS-over-HTTPS，看来项目前途无量啊！

### DISCUSSION THREAD

**Discussion by jzp820927** [SeaHOH/GotoX#10] [点子]
添加了一个很不错的附加 DNS 功能。顺便再提 2个建议：
1、是否能增加对 dns.google.com 增加自定义绑定 google IP 的配置项，这样对于手里有稳定 IP 的人来说，自己为 dns.google.com 指定 IP，以增加 dns 的性能和稳定。
2、ini 中没看到 DNS 有 listen 的选项，也就是说，这个 DNS 只能用于程序内部？是否可以考虑增加监听配置项，做成本地的小型 DNS 服务器，也可以为操作系统所用，不然有点太可惜了。

**Comment by SeaHOH** [comment]
就是觉得是个很好的备胎。 :laughing: 

1、可在 `Config.ini` 中配置 `overhttpslist`，指定 IP 列表名称，留空使用 `google_gws`。不会影响改变 dns.google.com 相关规则。
由于不支持 HTTP2 ，IP 数不要少了，不然请求数多的时候会很慢。我试过响应大多在 200-400 毫秒，偶尔有接近或刚刚超出 1000 毫秒。座标西南。

2、前段时间去掉了，不打算做这个。专门跟 DNS 磕的工具已经很多了。

**Comment by SeaHOH** [comment]
不好意思，上面第一条回复错了。
想起我是从内部请求 dns.google.com 的，没匹配规则，等会儿加上。

**Comment by jzp820927** [comment]
请求的域名应该是在 GAE 远端解析吧，如果仅仅只用于程序内部的话，还没想到是用在哪儿？莫非是用于 ActionFilter.ini 里面的 直连 和 转发 规则？

**Comment by SeaHOH** [comment]
现在可以自定义了，见上面修改的回复。

考虑了一下，这个设定还是不要影响全局的好，多花了点时间修改。

不是光用 GAE，还可以直连和转发 ，就是这样。
还有就是总是由 GAE 远程解析，发过去的永远是域名，除非没有域名。

**Comment by jzp820927** [comment]
由于不支持 HTTP2 ，IP 数不要少了，不然请求数多的时候会很慢。我试过响应大多在 200-400 毫秒，偶尔有接近或刚刚超出 1000 毫秒。座标西南。

关于你说的这个问题，其实是有解决办法的，就是不知道你想不想做（做不做你自己决定了），就是加 DNS 缓存，性能应该成倍提高，这也是我想提的第 3 个建议。

**Comment by SeaHOH** [comment]
缓存一直有，刚加了选项可以设置。

**Comment by jzp820927** [comment]
好，那基本上就差不多了，设置的话，主要就是缓存时间吧。==我试试

**Comment by jzp820927** [comment]
看来 DNS-over-HTTPS 主要是针对 ssl 这块了，默认的 sslaction 转发规则 + DNS-over-HTTPS ，至少 60%-70% 的访问请求可以跳过 GAE 了，对 GAE 的依赖也可以减轻很多，优势还是比较明显。

**Comment by caofengzi** [comment]
這個東西不支持32位win7？？？？？？？？

**Comment by SeaHOH** [comment]
@caofengzi 
便携版分支为 Win-64 位，32 位系统要自己安装 32 位的 Python3、 pyOpenSSL 和 gevent。
然后直接下载源码，运行 start.py 即可。

先前因为没啥人用这个，既然有需求今晚或明天就加上。

**Comment by jzp820927** [comment]
主要是你太低调了，所以知道的人少，自然用的人少。

**Comment by SeaHOH** [comment]
社长说得好，这样没压力，爱咋搞咋搞。 :smile:

**Comment by jzp820927** [comment]
这样也好，不然一天管理 ISSUES，都要弄到手软。

**Comment by SeaHOH** [comment]
@caofengzi 
请到 **win_py35** 分支下载 32 位便携版压缩包，用里面的内容完全替换 python 子目录即可。
https://github.com/SeaHOH/GotoX/blob/win_py35

现在的便携版还不完美，不方便升级，先弄个临时的。
